// ngx-barcode-scanner/bundles/ngx-barcode-scanner.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/common'), require('@angular/core'), require('@ericblade/quagga2'), require('lodash.defaultsdeep')) :
    typeof define === 'function' && define.amd ? define('ngx-barcode-scanner', ['exports', '@angular/common', '@angular/core', '@ericblade/quagga2', 'lodash.defaultsdeep'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory(global['ngx-barcode-scanner'] = {}, global.ng.common, global.ng.core, global.Quagga, global.defaultsDeep));
}(this, (function (exports, common, core, Quagga, defaultsDeep) { 'use strict';

    function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

    var Quagga__default = /*#__PURE__*/_interopDefaultLegacy(Quagga);
    var defaultsDeep__default = /*#__PURE__*/_interopDefaultLegacy(defaultsDeep);

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (Object.prototype.hasOwnProperty.call(b, p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
    function __rest(s, e) {
        var t = {};
        for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }
    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
            r = Reflect.decorate(decorators, target, key, desc);
        else
            for (var i = decorators.length - 1; i >= 0; i--)
                if (d = decorators[i])
                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }
    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); };
    }
    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
            return Reflect.metadata(metadataKey, metadataValue);
    }
    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }
    var __createBinding = Object.create ? (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        Object.defineProperty(o, k2, { enumerable: true, get: function () { return m[k]; } });
    }) : (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        o[k2] = m[k];
    });
    function __exportStar(m, o) {
        for (var p in m)
            if (p !== "default" && !Object.prototype.hasOwnProperty.call(o, p))
                __createBinding(o, m, p);
    }
    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m)
            return m.call(o);
        if (o && typeof o.length === "number")
            return {
                next: function () {
                    if (o && i >= o.length)
                        o = void 0;
                    return { value: o && o[i++], done: !o };
                }
            };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }
    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m)
            return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done)
                ar.push(r.value);
        }
        catch (error) {
            e = { error: error };
        }
        finally {
            try {
                if (r && !r.done && (m = i["return"]))
                    m.call(i);
            }
            finally {
                if (e)
                    throw e.error;
            }
        }
        return ar;
    }
    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }
    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++)
            s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    }
    ;
    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }
    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n])
            i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try {
            step(g[n](v));
        }
        catch (e) {
            settle(q[0][3], e);
        } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length)
            resume(q[0][0], q[0][1]); }
    }
    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }
    function __asyncValues(o) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function (v) { resolve({ value: v, done: d }); }, reject); }
    }
    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) {
            Object.defineProperty(cooked, "raw", { value: raw });
        }
        else {
            cooked.raw = raw;
        }
        return cooked;
    }
    ;
    var __setModuleDefault = Object.create ? (function (o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function (o, v) {
        o["default"] = v;
    };
    function __importStar(mod) {
        if (mod && mod.__esModule)
            return mod;
        var result = {};
        if (mod != null)
            for (var k in mod)
                if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k))
                    __createBinding(result, mod, k);
        __setModuleDefault(result, mod);
        return result;
    }
    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }
    function __classPrivateFieldGet(receiver, privateMap) {
        if (!privateMap.has(receiver)) {
            throw new TypeError("attempted to get private field on non-instance");
        }
        return privateMap.get(receiver);
    }
    function __classPrivateFieldSet(receiver, privateMap, value) {
        if (!privateMap.has(receiver)) {
            throw new TypeError("attempted to set private field on non-instance");
        }
        privateMap.set(receiver, value);
        return value;
    }

    var DEFAULT_CONFIG = {
        inputStream: {
            name: 'Live',
            type: 'LiveStream',
            target: null,
            constraints: {
                width: { min: 640 },
                height: { min: 480 },
                aspectRatio: { min: 1, max: 2 },
                facingMode: 'environment',
            },
            singleChannel: false // true: only the red color-channel is read
        },
        locator: {
            patchSize: 'medium',
            halfSample: true
        },
        locate: true,
        numOfWorkers: 4,
        decoder: {
            readers: ['code_128_reader']
        }
    };

    var BARCODE_TYPES = [
        'code_128',
        'code_39',
        'code_39_vin',
        'ean',
        'ean_extended',
        'ean_8',
        'upc',
        'upc_e',
        'codabar',
        'i2of5',
        '2of5',
        'code_93'
    ];
    function mapToReader(value) {
        if (typeof value === 'string') {
            checkBarCodeType(value);
            return [mapToBarcodeType(value)];
        }
        else {
            return value.map(function (val) {
                checkBarCodeType(val);
                return mapToBarcodeType(val);
            });
        }
    }
    function checkBarCodeType(value) {
        if (!BARCODE_TYPES.some(function (t) { return t === value; })) {
            throw new Error("This barcode type '" + value + "' is not valid.");
        }
    }
    function mapToBarcodeType(value) {
        return value + "_reader";
    }

    var BarcodeScannerLivestreamComponent = /** @class */ (function () {
        function BarcodeScannerLivestreamComponent() {
            this.maxWidth = '100%';
            // Outputs
            this.valueChanges = new core.EventEmitter();
            this.started = new core.EventEmitter();
            this._started = false;
        }
        Object.defineProperty(BarcodeScannerLivestreamComponent.prototype, "_maxWidth", {
            get: function () {
                return this.maxWidth ? "" + this.maxWidth : 'auto';
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(BarcodeScannerLivestreamComponent.prototype, "_maxHeight", {
            get: function () {
                return this.maxHeight ? "" + this.maxHeight : 'auto';
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(BarcodeScannerLivestreamComponent.prototype, "isStarted", {
            get: function () {
                return this._started;
            },
            enumerable: false,
            configurable: true
        });
        BarcodeScannerLivestreamComponent.prototype.ngOnDestroy = function () {
            this.stop();
        };
        BarcodeScannerLivestreamComponent.prototype.ngOnChanges = function (changes) {
            this.restart();
        };
        BarcodeScannerLivestreamComponent.prototype._init = function () {
            var _this = this;
            return new Promise(function (resolve, reject) {
                Quagga__default['default'].onProcessed(function (result) { return _this.onProcessed(result); });
                Quagga__default['default'].onDetected(function (result) { return _this.onDetected(result); });
                _this.configQuagga = defaultsDeep__default['default']({}, _this.config, DEFAULT_CONFIG);
                _this.configQuagga.inputStream.target = _this.barcodeScanner.nativeElement;
                if (_this.type) {
                    _this.configQuagga.decoder.readers = mapToReader(_this.type);
                }
                if (_this.deviceId) {
                    _this.configQuagga.inputStream.constraints.deviceId = _this.deviceId;
                }
                Quagga__default['default'].init(_this.configQuagga, function (err) {
                    if (err) {
                        console.log(err);
                        return reject(err);
                    }
                    resolve();
                });
            });
        };
        BarcodeScannerLivestreamComponent.prototype.start = function () {
            return __awaiter(this, void 0, void 0, function () {
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0:
                            if (!!this._started) return [3 /*break*/, 2];
                            return [4 /*yield*/, this._init()];
                        case 1:
                            _a.sent();
                            Quagga__default['default'].start();
                            this._started = true;
                            this.started.next(true);
                            _a.label = 2;
                        case 2: return [2 /*return*/];
                    }
                });
            });
        };
        BarcodeScannerLivestreamComponent.prototype.stop = function () {
            if (this._started) {
                Quagga__default['default'].stop();
                this._started = false;
                this.started.next(false);
            }
        };
        BarcodeScannerLivestreamComponent.prototype.restart = function () {
            if (this._started) {
                this.stop();
                this.start();
            }
        };
        BarcodeScannerLivestreamComponent.prototype.onProcessed = function (result) {
            var drawingCtx = Quagga__default['default'].canvas.ctx.overlay;
            var drawingCanvas = Quagga__default['default'].canvas.dom.overlay;
            if (result) {
                if (result.boxes) {
                    drawingCtx.clearRect(0, 0, parseInt(drawingCanvas.getAttribute('width'), 10), parseInt(drawingCanvas.getAttribute('height'), 10));
                    result.boxes.filter(function (box) {
                        return box !== result.box;
                    }).forEach(function (box) {
                        Quagga__default['default'].ImageDebug.drawPath(box, {
                            x: 0,
                            y: 1,
                        }, drawingCtx, {
                            color: 'green',
                            lineWidth: 2,
                        });
                    });
                }
                if (result.box) {
                    Quagga__default['default'].ImageDebug.drawPath(result.box, {
                        x: 0,
                        y: 1,
                    }, drawingCtx, {
                        color: '#00F',
                        lineWidth: 2,
                    });
                }
                if (result.codeResult && result.codeResult.code) {
                    Quagga__default['default'].ImageDebug.drawPath(result.line, {
                        x: 'x',
                        y: 'y',
                    }, drawingCtx, {
                        color: 'red',
                        lineWidth: 3,
                    });
                }
            }
        };
        BarcodeScannerLivestreamComponent.prototype.onDetected = function (result) {
            this.valueChanges.next(result);
        };
        return BarcodeScannerLivestreamComponent;
    }());
    BarcodeScannerLivestreamComponent.decorators = [
        { type: core.Component, args: [{
                    selector: 'barcode-scanner-livestream',
                    template: "<div\n  #BarcodeScanner\n  class=\"scanner\"\n  [hidden]=\"!isStarted\"\n  [style.max-height]=\"_maxHeight\"\n  [style.max-width]=\"_maxWidth\"\n>\n  <video [style.max-height]=\"_maxHeight\" [style.max-width]=\"_maxWidth\"></video>\n  <canvas\n    [style.max-height]=\"_maxHeight\"\n    [style.max-width]=\"_maxWidth\"\n    class=\"drawingBuffer\"\n  ></canvas>\n</div>\n",
                    encapsulation: core.ViewEncapsulation.None,
                    styles: [".scanner{position:relative}.scanner canvas,.scanner video{height:100%;width:100%}.scanner canvas.drawingBuffer{left:0;position:absolute;top:0}"]
                },] }
    ];
    BarcodeScannerLivestreamComponent.propDecorators = {
        type: [{ type: core.Input }],
        deviceId: [{ type: core.Input }],
        maxWidth: [{ type: core.Input }],
        maxHeight: [{ type: core.Input }],
        config: [{ type: core.Input }],
        valueChanges: [{ type: core.Output }],
        started: [{ type: core.Output }],
        barcodeScanner: [{ type: core.ViewChild, args: ['BarcodeScanner',] }]
    };

    var BarcodeScannerLivestreamModule = /** @class */ (function () {
        function BarcodeScannerLivestreamModule() {
        }
        return BarcodeScannerLivestreamModule;
    }());
    BarcodeScannerLivestreamModule.decorators = [
        { type: core.NgModule, args: [{
                    imports: [
                        common.CommonModule
                    ],
                    declarations: [
                        BarcodeScannerLivestreamComponent
                    ],
                    exports: [
                        BarcodeScannerLivestreamComponent
                    ]
                },] }
    ];

    var BarcodeScannerLivestreamOverlayComponent = /** @class */ (function () {
        function BarcodeScannerLivestreamOverlayComponent() {
            this._started = false;
            this.width = '90vw';
            this.maxWidth = '640px';
            this.valueChanges = new core.EventEmitter();
            this.started = new core.EventEmitter();
            this._showScanner = false;
        }
        Object.defineProperty(BarcodeScannerLivestreamOverlayComponent.prototype, "isStarted", {
            get: function () {
                return this._started;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(BarcodeScannerLivestreamOverlayComponent.prototype, "showScanner", {
            get: function () {
                return this._showScanner;
            },
            enumerable: false,
            configurable: true
        });
        BarcodeScannerLivestreamOverlayComponent.prototype.ngOnDestroy = function () {
            this.scanner.stop();
        };
        BarcodeScannerLivestreamOverlayComponent.prototype.show = function () {
            this._showScanner = true;
            this.scanner.start();
        };
        BarcodeScannerLivestreamOverlayComponent.prototype.hide = function () {
            this._showScanner = false;
            this.scanner.stop();
        };
        BarcodeScannerLivestreamOverlayComponent.prototype.onStarted = function (value) {
            this._started = value;
            this.started.next(value);
        };
        BarcodeScannerLivestreamOverlayComponent.prototype.onValueChanges = function (result) {
            this.valueChanges.next(result);
        };
        return BarcodeScannerLivestreamOverlayComponent;
    }());
    BarcodeScannerLivestreamOverlayComponent.decorators = [
        { type: core.Component, args: [{
                    selector: 'barcode-scanner-livestream-overlay',
                    template: "<div class=\"barcode-scanner-livestream-overlay\" [hidden]=\"!showScanner\">\n  <div\n    class=\"barcode-scanner-livestream-overlay-content\"\n    [style.width]=\"width\"\n    [style.max-width]=\"maxWidth\"\n    [style.height]=\"height\"\n    [style.max-height]=\"maxHeight\"\n  >\n    <div\n      class=\"barcode-scanner-livestream-overlay-close\"\n      *ngIf=\"isStarted\"\n      (click)=\"hide()\"\n    >\n      X\n    </div>\n    <barcode-scanner-livestream\n      [type]=\"type\"\n      [deviceId]=\"deviceId\"\n      [config]=\"config\"\n      (valueChanges)=\"onValueChanges($event)\"\n      (started)=\"onStarted($event)\"\n    ></barcode-scanner-livestream>\n  </div>\n</div>\n",
                    styles: [".barcode-scanner-livestream-overlay{background-color:rgba(0,0,0,.3);bottom:0;left:0;overflow:hidden;position:fixed;right:0;top:0;width:100%;z-index:1000}.barcode-scanner-livestream-overlay .barcode-scanner-livestream-overlay-content{left:50%;position:absolute;top:50%;transform:translate(-50%,-50%)}.barcode-scanner-livestream-overlay .barcode-scanner-livestream-overlay-content .barcode-scanner-livestream-overlay-close{background-color:#fff;border:2px solid #000;border-radius:2rem;box-sizing:initial;cursor:pointer;font-size:1.3rem;height:1.5rem;line-height:1.5rem;margin:-1rem;padding:.5rem;position:absolute;right:0;text-align:center;width:1.5rem;z-index:100}"]
                },] }
    ];
    BarcodeScannerLivestreamOverlayComponent.propDecorators = {
        type: [{ type: core.Input }],
        deviceId: [{ type: core.Input }],
        width: [{ type: core.Input }],
        maxWidth: [{ type: core.Input }],
        height: [{ type: core.Input }],
        maxHeight: [{ type: core.Input }],
        config: [{ type: core.Input }],
        valueChanges: [{ type: core.Output }],
        started: [{ type: core.Output }],
        scanner: [{ type: core.ViewChild, args: [BarcodeScannerLivestreamComponent,] }]
    };

    var BarcodeScannerLivestreamOverlayModule = /** @class */ (function () {
        function BarcodeScannerLivestreamOverlayModule() {
        }
        return BarcodeScannerLivestreamOverlayModule;
    }());
    BarcodeScannerLivestreamOverlayModule.decorators = [
        { type: core.NgModule, args: [{
                    imports: [
                        common.CommonModule,
                        BarcodeScannerLivestreamModule
                    ],
                    declarations: [
                        BarcodeScannerLivestreamOverlayComponent
                    ],
                    exports: [
                        BarcodeScannerLivestreamOverlayComponent
                    ]
                },] }
    ];

    /*
     * Public API Surface of ngx-barcode-scanner
     */

    /**
     * Generated bundle index. Do not edit.
     */

    exports.BarcodeScannerLivestreamComponent = BarcodeScannerLivestreamComponent;
    exports.BarcodeScannerLivestreamModule = BarcodeScannerLivestreamModule;
    exports.BarcodeScannerLivestreamOverlayComponent = BarcodeScannerLivestreamOverlayComponent;
    exports.BarcodeScannerLivestreamOverlayModule = BarcodeScannerLivestreamOverlayModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
