// ngx-barcode-scanner/esm2015/lib/barcode-scanner-livestream-overlay/barcode-scanner-livestream-overlay.component.js
import { Component, EventEmitter, Input, Output, ViewChild } from '@angular/core';
import { BarcodeScannerLivestreamComponent } from '../barcode-scanner-livestream/barcode-scanner-livestream.component';
export class BarcodeScannerLivestreamOverlayComponent {
    constructor() {
        this._started = false;
        this.width = '90vw';
        this.maxWidth = '640px';
        this.valueChanges = new EventEmitter();
        this.started = new EventEmitter();
        this._showScanner = false;
    }
    get isStarted() {
        return this._started;
    }
    get showScanner() {
        return this._showScanner;
    }
    ngOnDestroy() {
        this.scanner.stop();
    }
    show() {
        this._showScanner = true;
        this.scanner.start();
    }
    hide() {
        this._showScanner = false;
        this.scanner.stop();
    }
    onStarted(value) {
        this._started = value;
        this.started.next(value);
    }
    onValueChanges(result) {
        this.valueChanges.next(result);
    }
}
BarcodeScannerLivestreamOverlayComponent.decorators = [
    { type: Component, args: [{
                selector: 'barcode-scanner-livestream-overlay',
                template: "<div class=\"barcode-scanner-livestream-overlay\" [hidden]=\"!showScanner\">\n  <div\n    class=\"barcode-scanner-livestream-overlay-content\"\n    [style.width]=\"width\"\n    [style.max-width]=\"maxWidth\"\n    [style.height]=\"height\"\n    [style.max-height]=\"maxHeight\"\n  >\n    <div\n      class=\"barcode-scanner-livestream-overlay-close\"\n      *ngIf=\"isStarted\"\n      (click)=\"hide()\"\n    >\n      X\n    </div>\n    <barcode-scanner-livestream\n      [type]=\"type\"\n      [deviceId]=\"deviceId\"\n      [config]=\"config\"\n      (valueChanges)=\"onValueChanges($event)\"\n      (started)=\"onStarted($event)\"\n    ></barcode-scanner-livestream>\n  </div>\n</div>\n",
                styles: [".barcode-scanner-livestream-overlay{background-color:rgba(0,0,0,.3);bottom:0;left:0;overflow:hidden;position:fixed;right:0;top:0;width:100%;z-index:1000}.barcode-scanner-livestream-overlay .barcode-scanner-livestream-overlay-content{left:50%;position:absolute;top:50%;transform:translate(-50%,-50%)}.barcode-scanner-livestream-overlay .barcode-scanner-livestream-overlay-content .barcode-scanner-livestream-overlay-close{background-color:#fff;border:2px solid #000;border-radius:2rem;box-sizing:initial;cursor:pointer;font-size:1.3rem;height:1.5rem;line-height:1.5rem;margin:-1rem;padding:.5rem;position:absolute;right:0;text-align:center;width:1.5rem;z-index:100}"]
            },] }
];
BarcodeScannerLivestreamOverlayComponent.propDecorators = {
    type: [{ type: Input }],
    deviceId: [{ type: Input }],
    width: [{ type: Input }],
    maxWidth: [{ type: Input }],
    height: [{ type: Input }],
    maxHeight: [{ type: Input }],
    config: [{ type: Input }],
    valueChanges: [{ type: Output }],
    started: [{ type: Output }],
    scanner: [{ type: ViewChild, args: [BarcodeScannerLivestreamComponent,] }]
};
