// ngx-barcode-scanner/fesm2015/ngx-barcode-scanner.js
import { CommonModule } from '@angular/common';
import { EventEmitter, Component, ViewEncapsulation, Input, Output, ViewChild, NgModule } from '@angular/core';
import { __awaiter } from 'tslib';
import Quagga from '@ericblade/quagga2';
import defaultsDeep from 'lodash.defaultsdeep';

const DEFAULT_CONFIG = {
    inputStream: {
        name: 'Live',
        type: 'LiveStream',
        target: null,
        constraints: {
            width: { min: 640 },
            height: { min: 480 },
            aspectRatio: { min: 1, max: 2 },
            facingMode: 'environment',
        },
        singleChannel: false // true: only the red color-channel is read
    },
    locator: {
        patchSize: 'medium',
        halfSample: true
    },
    locate: true,
    numOfWorkers: 4,
    decoder: {
        readers: ['code_128_reader']
    }
};

const BARCODE_TYPES = [
    'code_128',
    'code_39',
    'code_39_vin',
    'ean',
    'ean_extended',
    'ean_8',
    'upc',
    'upc_e',
    'codabar',
    'i2of5',
    '2of5',
    'code_93'
];
function mapToReader(value) {
    if (typeof value === 'string') {
        checkBarCodeType(value);
        return [mapToBarcodeType(value)];
    }
    else {
        return value.map(val => {
            checkBarCodeType(val);
            return mapToBarcodeType(val);
        });
    }
}
function checkBarCodeType(value) {
    if (!BARCODE_TYPES.some(t => t === value)) {
        throw new Error(`This barcode type '${value}' is not valid.`);
    }
}
function mapToBarcodeType(value) {
    return `${value}_reader`;
}

class BarcodeScannerLivestreamComponent {
    constructor() {
        this.maxWidth = '100%';
        // Outputs
        this.valueChanges = new EventEmitter();
        this.started = new EventEmitter();
        this._started = false;
    }
    get _maxWidth() {
        return this.maxWidth ? `${this.maxWidth}` : 'auto';
    }
    get _maxHeight() {
        return this.maxHeight ? `${this.maxHeight}` : 'auto';
    }
    get isStarted() {
        return this._started;
    }
    ngOnDestroy() {
        this.stop();
    }
    ngOnChanges(changes) {
        this.restart();
    }
    _init() {
        return new Promise((resolve, reject) => {
            Quagga.onProcessed((result) => this.onProcessed(result));
            Quagga.onDetected((result) => this.onDetected(result));
            this.configQuagga = defaultsDeep({}, this.config, DEFAULT_CONFIG);
            this.configQuagga.inputStream.target = this.barcodeScanner.nativeElement;
            if (this.type) {
                this.configQuagga.decoder.readers = mapToReader(this.type);
            }
            if (this.deviceId) {
                this.configQuagga.inputStream.constraints.deviceId = this.deviceId;
            }
            Quagga.init(this.configQuagga, (err) => {
                if (err) {
                    console.log(err);
                    return reject(err);
                }
                resolve();
            });
        });
    }
    start() {
        return __awaiter(this, void 0, void 0, function* () {
            if (!this._started) {
                yield this._init();
                Quagga.start();
                this._started = true;
                this.started.next(true);
            }
        });
    }
    stop() {
        if (this._started) {
            Quagga.stop();
            this._started = false;
            this.started.next(false);
        }
    }
    restart() {
        if (this._started) {
            this.stop();
            this.start();
        }
    }
    onProcessed(result) {
        const drawingCtx = Quagga.canvas.ctx.overlay;
        const drawingCanvas = Quagga.canvas.dom.overlay;
        if (result) {
            if (result.boxes) {
                drawingCtx.clearRect(0, 0, parseInt(drawingCanvas.getAttribute('width'), 10), parseInt(drawingCanvas.getAttribute('height'), 10));
                result.boxes.filter((box) => {
                    return box !== result.box;
                }).forEach((box) => {
                    Quagga.ImageDebug.drawPath(box, {
                        x: 0,
                        y: 1,
                    }, drawingCtx, {
                        color: 'green',
                        lineWidth: 2,
                    });
                });
            }
            if (result.box) {
                Quagga.ImageDebug.drawPath(result.box, {
                    x: 0,
                    y: 1,
                }, drawingCtx, {
                    color: '#00F',
                    lineWidth: 2,
                });
            }
            if (result.codeResult && result.codeResult.code) {
                Quagga.ImageDebug.drawPath(result.line, {
                    x: 'x',
                    y: 'y',
                }, drawingCtx, {
                    color: 'red',
                    lineWidth: 3,
                });
            }
        }
    }
    onDetected(result) {
        this.valueChanges.next(result);
    }
}
BarcodeScannerLivestreamComponent.decorators = [
    { type: Component, args: [{
                selector: 'barcode-scanner-livestream',
                template: "<div\n  #BarcodeScanner\n  class=\"scanner\"\n  [hidden]=\"!isStarted\"\n  [style.max-height]=\"_maxHeight\"\n  [style.max-width]=\"_maxWidth\"\n>\n  <video [style.max-height]=\"_maxHeight\" [style.max-width]=\"_maxWidth\"></video>\n  <canvas\n    [style.max-height]=\"_maxHeight\"\n    [style.max-width]=\"_maxWidth\"\n    class=\"drawingBuffer\"\n  ></canvas>\n</div>\n",
                encapsulation: ViewEncapsulation.None,
                styles: [".scanner{position:relative}.scanner canvas,.scanner video{height:100%;width:100%}.scanner canvas.drawingBuffer{left:0;position:absolute;top:0}"]
            },] }
];
BarcodeScannerLivestreamComponent.propDecorators = {
    type: [{ type: Input }],
    deviceId: [{ type: Input }],
    maxWidth: [{ type: Input }],
    maxHeight: [{ type: Input }],
    config: [{ type: Input }],
    valueChanges: [{ type: Output }],
    started: [{ type: Output }],
    barcodeScanner: [{ type: ViewChild, args: ['BarcodeScanner',] }]
};

class BarcodeScannerLivestreamModule {
}
BarcodeScannerLivestreamModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    BarcodeScannerLivestreamComponent
                ],
                exports: [
                    BarcodeScannerLivestreamComponent
                ]
            },] }
];

class BarcodeScannerLivestreamOverlayComponent {
    constructor() {
        this._started = false;
        this.width = '90vw';
        this.maxWidth = '640px';
        this.valueChanges = new EventEmitter();
        this.started = new EventEmitter();
        this._showScanner = false;
    }
    get isStarted() {
        return this._started;
    }
    get showScanner() {
        return this._showScanner;
    }
    ngOnDestroy() {
        this.scanner.stop();
    }
    show() {
        this._showScanner = true;
        this.scanner.start();
    }
    hide() {
        this._showScanner = false;
        this.scanner.stop();
    }
    onStarted(value) {
        this._started = value;
        this.started.next(value);
    }
    onValueChanges(result) {
        this.valueChanges.next(result);
    }
}
BarcodeScannerLivestreamOverlayComponent.decorators = [
    { type: Component, args: [{
                selector: 'barcode-scanner-livestream-overlay',
                template: "<div class=\"barcode-scanner-livestream-overlay\" [hidden]=\"!showScanner\">\n  <div\n    class=\"barcode-scanner-livestream-overlay-content\"\n    [style.width]=\"width\"\n    [style.max-width]=\"maxWidth\"\n    [style.height]=\"height\"\n    [style.max-height]=\"maxHeight\"\n  >\n    <div\n      class=\"barcode-scanner-livestream-overlay-close\"\n      *ngIf=\"isStarted\"\n      (click)=\"hide()\"\n    >\n      X\n    </div>\n    <barcode-scanner-livestream\n      [type]=\"type\"\n      [deviceId]=\"deviceId\"\n      [config]=\"config\"\n      (valueChanges)=\"onValueChanges($event)\"\n      (started)=\"onStarted($event)\"\n    ></barcode-scanner-livestream>\n  </div>\n</div>\n",
                styles: [".barcode-scanner-livestream-overlay{background-color:rgba(0,0,0,.3);bottom:0;left:0;overflow:hidden;position:fixed;right:0;top:0;width:100%;z-index:1000}.barcode-scanner-livestream-overlay .barcode-scanner-livestream-overlay-content{left:50%;position:absolute;top:50%;transform:translate(-50%,-50%)}.barcode-scanner-livestream-overlay .barcode-scanner-livestream-overlay-content .barcode-scanner-livestream-overlay-close{background-color:#fff;border:2px solid #000;border-radius:2rem;box-sizing:initial;cursor:pointer;font-size:1.3rem;height:1.5rem;line-height:1.5rem;margin:-1rem;padding:.5rem;position:absolute;right:0;text-align:center;width:1.5rem;z-index:100}"]
            },] }
];
BarcodeScannerLivestreamOverlayComponent.propDecorators = {
    type: [{ type: Input }],
    deviceId: [{ type: Input }],
    width: [{ type: Input }],
    maxWidth: [{ type: Input }],
    height: [{ type: Input }],
    maxHeight: [{ type: Input }],
    config: [{ type: Input }],
    valueChanges: [{ type: Output }],
    started: [{ type: Output }],
    scanner: [{ type: ViewChild, args: [BarcodeScannerLivestreamComponent,] }]
};

class BarcodeScannerLivestreamOverlayModule {
}
BarcodeScannerLivestreamOverlayModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule,
                    BarcodeScannerLivestreamModule
                ],
                declarations: [
                    BarcodeScannerLivestreamOverlayComponent
                ],
                exports: [
                    BarcodeScannerLivestreamOverlayComponent
                ]
            },] }
];

/*
 * Public API Surface of ngx-barcode-scanner
 */

/**
 * Generated bundle index. Do not edit.
 */

export { BarcodeScannerLivestreamComponent, BarcodeScannerLivestreamModule, BarcodeScannerLivestreamOverlayComponent, BarcodeScannerLivestreamOverlayModule };
